### ISSUE TITLE
Attempt Uphold wallet relinking more often

### ISSUE THREAD

**Issue by zenparsing** [brave/brave-browser#14573]
Currently, Rewards will attempt to link the user's Uphold wallet to their payment ID only once for the user's profile. Instead, we should attempt to re-link the user's Uphold wallet when they successfully complete the Uphold login process.

**Comment by LaurenWags** [comment]
labelling as `QA/Blocked` until this is uplifted to 1.22.x and a new build is created

**Comment by kjozwiak** [comment]
Merged into `1.22.x` via https://github.com/brave/brave-core/pull/8211#pullrequestreview-615513502. Just need a new build which we'll get soon. Still waiting on one more PR.

**Comment by srirambv** [comment]
Verification passed on OnePlus 6T with Android 10 running 1.22.65 x64 build
- Verified disconnecting an existing wallet shows up on log
```
2021-03-19 08:37:33.213 6242-6242/? V/chromium: [VERBOSE1:uphold.cc(196)] Disconnecting wallet
```
- Verified reconnecting wallet shows the confirmation of existing wallet
```
2021-03-19 08:39:37.495 6242-6242/? V/chromium: [VERBOSE1:wallet_create.cc(35)] Wallet already exists
2021-03-19 08:39:37.502 6242-6242/? V/chromium: [VERBOSE5:ledger_impl.cc(133)] 
    [ REQUEST ]
    > URL: https://api-sandbox.uphold.com/oauth2/token
    > Method: UrlMethod::POST
    > Content: code=44..................&grant_type=authorization_code
    > Content Type: application/x-www-form-urlencoded
```
```
2021-03-19 08:34:21.836 6242-6242/? V/chromium: [VERBOSE5:ledger_impl.cc(133)] 
    [ REQUEST ]
    > URL: https://grant.rewards.bravesoftware.com/v3/wallet/uphold/7ba............................./claim
    > Method: UrlMethod::POST
    > Content: {"signedLinkingRequest":"eyJ.........."}
    > Content Type: application/json; charset=utf-8
```
- Verified able to reconnect/relink from panel as well
```
2021-03-19 08:43:07.550 6242-6242/? V/chromium: [VERBOSE6:logging_util.cc(136)] 
    [ RESPONSE - OnRequest ]
    > Url: https://grant.rewards.bravesoftware.com/v3/wallet/uphold/7ba.............................../claim
    > Result: Success
    > HTTP Code: 200
    > Body: null
```
